component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -484,7 +484,25 @@ private void processSMSOTPFlow(AuthenticationContext context, HttpServletRequest
                 String loginPage = SMSOTPUtils.getMobileNumberRequestPage(context);
                 try {
                     String url = getURL(loginPage, queryParams);
-                    response.sendRedirect(url);
+                    String mobileNumberPatternViolationIssue = SMSOTPConstants.MOBILE_NUMBER_PATTERN_POLICY_VIOLATED;
+                    if (StringUtils.isNotEmpty(
+                            context.getAuthenticatorProperties().get(SMSOTPConstants.MOBILE_NUMBER_REGEX))) {
+                        // Check for regex is violation error message configured in idp configuration.
+                        if (StringUtils.isNotEmpty(context.getAuthenticatorProperties()
+                                .get(SMSOTPConstants.MOBILE_NUMBER_PATTERN_FAILURE_ERROR_MESSAGE))) {
+                            mobileNumberPatternViolationIssue = context.getAuthenticatorProperties()
+                                    .get(SMSOTPConstants.MOBILE_NUMBER_PATTERN_FAILURE_ERROR_MESSAGE);
+                        }
+                        // Send the response with encoded regex pattern and error message.
+                        response.sendRedirect(url + SMSOTPConstants.MOBILE_NUMBER_REGEX_PATTERN_QUERY +
+                                getEncoder().encodeToString(
```

Comment:
does getEncoder().encodeToString handle URL encoding?

Reply:
We are decoding here.
https://github.com/wso2-support/identity-apps/pull/45/files#diff-702c9cc5c9eac5a6b39af1545a0f52fdbaafe39b974ca38ca392deb4418a8a0bR41

Existing usage: https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/blob/821f67cf399b05c609077724a959383622a17093/component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java#L613

Reply:
We are decoding here.
https://github.com/wso2-support/identity-apps/pull/45/files#diff-702c9cc5c9eac5a6b39af1545a0f52fdbaafe39b974ca38ca392deb4418a8a0bR41

Existing usage: https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/blob/821f67cf399b05c609077724a959383622a17093/component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java#L613